dali/c_api/c_api.cc

```diff
@@ -115,3 +116,11 @@ void daliDeletePipeline(daliPipelineHandle* pipe_handle) {
   pipe_handle->pipe = nullptr;
 }
 
+void daliLoadLibrary(const char* lib_path) {
+  try {
+    dali::PluginManager::Instance().LoadLibrary(lib_path);
+  } catch( const std::exception& e ) {
+    // TODO(janton): what to do when an error occurs?
```

Comment:
I think, a DALI_FAIL would do it here:
```
DALI_FAIL(dali::string("An exceptioned occured when trying the library:") + dali::string(e.what()));
```

Reply:
I'd pass on `try catch` block here. If the loading library failed, as a user I'd expect a failure anyhow. With `try catch` we're only rethrowing caught exception, while there's no need for additional mem cleaning or anything.

Reply:
we cannot throw an exception to a C caller, that was the point of catch block in the first place. I guess I will remove it and leave that for a refactoring of the c_api.cc (since it affects the rest of the file as well)

Reply:
Tracked as **DALI-446**